Portal backport patch4

sjpadgett opened the pull request:
#### Short description of what this resolves:
Bringing some LBF stuff back. Also various events support for telehealth

#### Changes proposed in this pull request:
added sortablejs to portal directory tree.
still testing


below added by bradymiller:
Up For Grab demo for this PR is here: https://www.open-emr.org/wiki/index.php/Development_Demo#Zeta_-_Up_For_Grabs_Demo

sjpadgett commented:
@sunsetsystems Hi Rod, I tried to get around bringing some of your LBF style changes back but I really need them for portal LBF templates. If you get a second, would you look over and see if you spot any gotcha's I may have missed. These mods have been in testing for a couple months between myself and at least three other sites. Still, if anybody will get bitten, it'd be me!

@adunsulag I think I have all your new event classes but i'm still looking over.

adunsulag commented:
@sjpadgett I'll try pulling your PR and testing it with my telehealth module tomorrow morning and I can see if there's anything missing.

bradymiller commented on sql/patch.sql:
```diff
@@ -147,3 +147,106 @@ KEY (`success`)
 ) ENGINE=InnoDB;
 #EndIf
```
also need to make these changes in database.sql

sjpadgett commented:
> I'll try pulling your PR and testing it with my telehealth module tomorrow morning and I can see if there's anything missing.

@adunsulag I added the event to main.php however, two others I'm unsure about.
- interface/usergroup/usergroup_admin.php
- PatientService was used in options.inc.php but I didn't include in backport. This is where you refactored search and added a var.
- I'm guessing calendar needs mods also.

adunsulag commented:
@sjpadgett I pulled down the PR and tried building it using the development-easy docker but the themes are not building for some reason.  I'm probably doing something wrong as I'm not used to working on the 6_0_0 release.  I'll see if I can figure it out.
```
openemr_1     | Running quick setup!
openemr_1     | PHP Warning:  Undefined array key 1 in /var/www/localhost/htdocs/auto_configure.php on line 31
openemr_1     | PHP Warning:  mysqli_real_connect(): (HY000/2002): Connection refused in /var/www/localhost/htdocs/openemr/library/classes/Installer.class.php on line 1240
openemr_1     | PHP Fatal error:  Uncaught Exception: ERROR: unable to connect to database as root
openemr_1     |  in /var/www/localhost/htdocs/auto_configure.php:47
openemr_1     | Stack trace:
openemr_1     | #0 {main}
openemr_1     |   thrown in /var/www/localhost/htdocs/auto_configure.php on line 47
openemr_1     | Couldn't set up. Any of these reasons could be what's wrong:
openemr_1     |  - You didn't spin up a MySQL container or connect your OpenEMR container to a mysql instance
openemr_1     |  - MySQL is still starting up and wasn't ready for connection yet
openemr_1     |  - The Mysql credentials were incorrect
mysql_1       | Warning: Unable to load '/usr/share/zoneinfo/tzdata.zi' as time zone. Skipping it.
```

bradymiller commented:
Up for grabs demo for this PR is here: https://www.open-emr.org/wiki/index.php/Development_Demo#Zeta_-_Up_For_Grabs_Demo

sjpadgett commented:
![image](https://user-images.githubusercontent.com/1859985/154144692-f76e79a9-1229-47be-b3ec-7a8c3621cf7a.png)

sjpadgett commented on src/Common/Uuid/UuidRegistry.php:
```diff
@@ -25,15 +25,41 @@
 {
-
+    const UUID_MAX_BATCH_COUNT = 1000;
+    const UUID_TABLE_DEFINITIONS = [
+        'ccda' => ['table_name' => 'ccda'],
+        'documents' => ['table_name' => 'documents'],
+        'drugs' => ['table_name' => 'drugs', 'table_id' => 'drug_id'],
+        'facility' => ['table_name' => 'facility'],
+        'facility_user_ids' => ['table_name' => 'facility_user_ids', 'table_vertical' => ['uid', 'facility_id']],
+        'form_clinical_notes' => ['table_name' => 'form_clinical_notes'],
+        'form_encounter' => ['table_name' => 'form_encounter'],
+        'form_vitals' => ['table_name' => 'form_vitals'],
+        'history_data' => ['table_name' => 'history_data'],
+        'immunizations' => ['table_name' => 'immunizations'],
+        'insurance_companies' => ['table_name' => 'insurance_companies'],
+        'insurance_data' => ['table_name' => 'insurance_data'],
+        'lists' => ['table_name' => 'lists'],
+        'patient_data' => ['table_name' => 'patient_data'],
+        'patient_history' => ['table_name' => 'patient_history'],
+        'prescriptions' => ['table_name' => 'prescriptions'],
+        'procedure_order' => ['table_name' => 'procedure_order', 'table_id' => 'procedure_order_id'],
+        'procedure_providers' => ['table_name' => 'procedure_providers', 'table_id' => 'ppid'],
+        'procedure_report' => ['table_name' => 'procedure_report', 'table_id' => 'procedure_report_id'],
+        'procedure_result' => ['table_name' => 'procedure_result', 'table_id' => 'procedure_result_id'],
+        'users' => ['table_name' => 'users']
+    ];
```
@adunsulag The following are not in v6 so, what do we need
       'documents' => ['table_name' => 'documents'],
        'form_vitals' => ['table_name' => 'form_vitals'],
        'history_data' => ['table_name' => 'history_data'],
        'form_clinical_notes' => ['table_name' => 'form_clinical_notes'],
        'insurance_companies' => ['table_name' => 'insurance_companies'],
        'patient_history' => ['table_name' => 'patient_history'],
        'insurance_data' => ['table_name' => 'insurance_data'],
        'procedure_providers' => ['table_name' => 'procedure_providers', 'table_id' => 'ppid'],
        'procedure_report' => ['table_name' => 'procedure_report', 'table_id' => 'procedure_report_id'],

adunsulag replied:
None of those are used by my telehealth stuff.  The only thing I use is users and patient_data.  So if they aren't in v6 I would just remove them.

sjpadgett commented:
closed with PR #4911 replacing.
